cypress/e2e/po/extensions/kubewarden/kubewarden.utils.ts

```diff
@@ -0,0 +1,245 @@
+    super(KubewardenPolicyServerListPagePo.url);
+  }
+}
+
+class KubewardenPolicyServerEditPagePo extends PagePo {
+  private static createPath(clusterId: string, id?: string ) {
+    const root = `/c/${ clusterId }/kubewarden/policies.kubewarden.io.policyserver`;
+
+    return id ? `${ root }/${ id }?mode=edit` : `${ root }/create`;
+  }
+
+  goTo(clusterId: string, id?:string): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(KubewardenPolicyServerEditPagePo.createPath(clusterId, id));
+  }
+
+  constructor(clusterId = 'local', id?: string) {
+    super(KubewardenPolicyServerEditPagePo.createPath(clusterId, id));
+  }
+
+  editYamlBtnClick():Cypress.Chainable {
+    return this.self().get('[data-testid="kw-policy-server-config-yaml-option"] [data-testid="button-group-child-1"]').click();
+  }
+
+  // the edit view of the policy server is not a generic resource edit/yaml view, so let's go
+  // straight to the source
+  codeMirror() {
+    return CodeMirrorPo.bySelector(this.self(), '[data-testid="yaml-editor-code-mirror"]');
+  }
+
+  saveForm(): AsyncButtonPo {
+    return new AsyncButtonPo('[data-testid="form-save"]', this.self());
+  }
+
+  waitForPolicyServerCreation(interceptName: string, nameToCheck: string) {
+    cy.wait(`@${ interceptName }`, { requestTimeout: 20000 }).then(({ response }) => {
+      expect(response?.statusCode).to.eq(201);
+      expect(response?.body.metadata).to.have.property('name', nameToCheck);
+
+      cy.wait(5000); // eslint-disable-line cypress/no-unnecessary-waiting
+    });
+  }
+}
+
+class KubewardenPolicyServerDetailPagePo extends PagePo {
+  private static createPath(clusterId: string, id: string) {
+    return `/c/${ clusterId }/kubewarden/policies.kubewarden.io.policyserver/${ id }`;
+  }
+
+  goTo(clusterId: string, id:string): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(KubewardenPolicyServerDetailPagePo.createPath(clusterId, id));
+  }
+
+  constructor(clusterId = 'local', id: string) {
+    super(KubewardenPolicyServerDetailPagePo.createPath(clusterId, id));
+  }
+
+  metricsAddServiceMonitorClick() {
+    this.self().getId('kw-monitoring-checklist-step-service-monitor-button').click();
+  }
+
+  metricsAddGrafanaDasboardClick() {
+    this.self().getId('kw-monitoring-checklist-step-config-map-button').click();
+  }
+}
+
+class AdmissionPoliciesListPo extends BaseResourceList {
+  state(name: string) {
+    return this.resourceTable().sortableTable().rowWithName(name).column(1);
+  }
+}
+
+class KubewardenAdmissionPoliciesListPagePo extends PagePo {
+  static url = '/c/local/kubewarden/policies.kubewarden.io.admissionpolicy';
+  static goTo(): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(KubewardenAdmissionPoliciesListPagePo.url);
+  }
+
+  constructor() {
+    super(KubewardenAdmissionPoliciesListPagePo.url);
+  }
+
+  addToArtifactHubClick() {
+    this.self().getId('action-button-async-button').click();
+  }
+
+  apOfficialPoliciesTable() {
```

Comment:
dupe of list?

Reply:
Nope. It's a custom list for the official  admission policy "templates". One can argue if it belongs to an `admissionPoliciesList` but it's also a bit strange because it's after you hit "create" on a normal list view, but it's actually a mid-way screen before the actual create interface.

I figured it didn't matter too much, but for sure it's not a normal list.

Reply:
That naming is slightly confusing. It's not only the "official" policy templates, it can be any policies within [ArtifactHub](https://artifacthub.io/packages/search?kind=13). We do filter by default the policies created by Kubewarden, but down the line if we want to test the filtering of this policy list it would be confusing with this naming scheme.